eng/SourceBuild.props

```diff
@@ -83,4 +83,35 @@
     </ItemGroup>
   </Target>
 
+  <Target Name="RestoreNpmPackages"
```

Comment:
What are the prereqs for this?  I am wondering if changes are going to be needed to the containers used in SB CI.  This should be flushed out to prevent blocking the dependency flow into installer.

Reply:
The image needs node 20 (the LTS) version to be part of the image.

Reply:
Only one of the container images used in SB CI has node - ubuntu 22.04 and it is extremely old.  The list is specified here - https://github.com/dotnet/installer/blob/main/eng/pipelines/templates/stages/vmr-build.yml#L19  Can someone from aspnet add node prior to this flowing into installer?

Regarding the version - has there been any discussion with @dotnet/distro-maintainers to ensure this is the minimum version supported across the distros that will source-build .NET 9.0?  @leecow this is something to raise in the next source-build partner sync.

Reply:
Thanks, @MichaelSimons !  Since this becomes a build-time dependency with minimum version, I agree on calling this out with distros and partners. After all, this impacts whether/where we can build .NET 9.

Hopefully, it shouldn't be an issue this time around.

https://github.com/nodejs/release#release-schedule is a good resource for Node.js versioning. The last version before 20 (18) goes EOL early 2025 - so just a few months after .NET 9 GA. I don't think using that is really an option for .NET 9, unless a rebase is on the table during a servicing release just a few months post GA. Even  Node.js 20 goes EOL in mid-2026, (almost) aligning with the expected .NET 9 EOL.

Reply:
@MichaelSimons we've already added Node to `centos-stream8`, as well as new alpine image - `alpine-3.19-WithNode`. Does every image in that list need to have node added?

Reply:
We need to test on each of these distros.  We can update to a newer version per what .NET 9 will support when it releases (e.g. updating to the new Alpine 3.19 image)

Reply:
Gotcha - so to make sure I understand:

 - alpine-3.17: can be updated to alpine-3.19-WithNode
 - centos-stream8: already has Node 20
 - centos-stream9, fedora-39, ubuntu-22.04. ubuntu-22.04-arm64 : need to add Node 20

Does that sound right? If so I can do that today.

Reply:
> Does that sound right? If so I can do that today.

Yes that sounds correct.

Reply:
https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/949